dashboards-observability/.cypress/integration/1_event_analytics.spec.js

```diff
@@ -615,4 +639,133 @@ describe('Renders data view', () => {
     cy.get('[data-test-subj="workspace__dataTableViewSwitch"]').click();
     cy.get('[data-test-subj="workspace__dataTable"]').should('not.exist');
   });
-});
\ No newline at end of file
+});
+
+describe('Renders Tree Map', () => {
+  beforeEach(() => {
+    landOnEventVisualizations();
+  });
+
+  it('Renders Tree Map', () => {
+    renderTreeMapchart();
+    cy.get('.euiFlexItem.euiFlexItem--flexGrowZero .euiButton__text').eq(2).click();
+    cy.get('path.surface').should('have.length', 176);
+  });
+
+  it('Renders Tree Map, add value parameters and verify Reset button click is working', () => {
+   renderTreeMapchart();
+    cy.get('.euiFlexItem.euiFlexItem--flexGrowZero .euiButton__text').eq(2).click();
+    cy.get('[data-test-subj="visualizeEditorResetButton"]').click();
+    cy.get('#configPanel__panelOptions .euiFieldText').should('have.value', '');
+    cy.get('.euiFlexItem .euiFormRow [placeholder="Description"]').should('have.value', '');
+    cy.get('.euiComboBox__inputWrap.euiComboBox__inputWrap-isClearable').eq(1).should('have.value', '');
+    cy.get('.euiComboBox__inputWrap.euiComboBox__inputWrap-isClearable').eq(2).should('have.value', '');
+    cy.get('.euiComboBox__inputWrap.euiComboBox__inputWrap-isClearable').eq(3).should('have.value', '');
+  });
+
+  it('Renders Tree Map, Save and Delete Visualization', () => {
+    renderTreeMapchart();
+    cy.get('.euiFlexItem.euiFlexItem--flexGrowZero .euiButton__text').eq(2).click();
+    saveVisualizationAndVerify();
+    cy.wait(delay * 4);
+    deleteVisualization();
+  });
+  
+  it('Render Tree Map chart and verify color theme under Chart styles options', () =>{
+    renderTreeMapchart();
+    cy.get('.euiTitle.euiTitle--xxsmall').contains('Color Theme').should('exist');
+    cy.get('.euiSuperSelectControl').contains('Default').click();
+    cy.get('.euiContextMenuItem__text .euiColorPalettePicker__item').eq(1).contains('Single color').click();
+    cy.get('.euiFieldText.euiColorPicker__input.euiFieldText--withIcon').click();
+    cy.get('[aria-label="Select #D36086 as the color"]').click();
+    cy.get('.euiButton__text').contains('Preview').should('exist').click();
+    cy.get('path[style*="rgb(29, 30, 36)"]').eq(0).should('exist');
```

Comment:
This RGB values can get tricky sometimes. If a different browser renders the color a bit differently the values can come different. Do we really need to check this color here? Did we define these RGB values or these are system generated?

Reply:
Yes Subrat, those values are system generated. But we don't have any other attribute to reverify if the color which we have selected is appearing right in the chart. That is why I have used RGB value in assertion.